src/PassType.cpp

```diff
@@ -0,0 +1,403 @@
+                if (allowedRedefValue == "true") {
+                    ALLOW_REDEFINE = true; 
+                    break;
+                }
+                else if (allowedRedefValue == "false") {
+                    ALLOW_REDEFINE = false; 
+                    break; 
+                }
+            }
+        }
+
+    }
+    testFile.close(); 
+
+    return ALLOW_REDEFINE; 
+}
+
+
+void PassType::transfer(Instruction* instruction, SetVector<Instruction *> workSet, std::map<std::string, MaybeUninitMethodsSet>& inputMethodsSet) {
+   if (auto Alloca = dyn_cast<AllocaInst>(instruction)) {
+    /*
+    Typically allocation instructions are used for local identifiers which start with a %, so it's fine to add a % to the name here 
+    */
+    inputMethodsSet["%" + Alloca->getName().str()] = {};
+    
+  }
+  else if (auto Store = dyn_cast<StoreInst>(instruction)) {
+
+    Value* valueToStore = Store->getOperand(0);      
+    Value* receivingValue = Store->getOperand(1);   
+ 
+    for (auto Inst : workSet) {
+      if (valueToStore == Inst) {
+        if (auto Call = dyn_cast<CallInst>(Inst)) {
+          std::string argName = dataflow::variable(Store->getOperand(1)); 
+
+          if (argName[0] == '@') {
+            argName[0] = '%'; 
+          }
+
+          std::string fnName = Call->getCalledFunction()->getName().str(); 
+
+
+          if (
+            !this->unsafeFunctions.count(fnName)  && 
+            !this->unsafeFunctions.count(this->memoryFunctions[fnName]) && 
+            this->memoryFunctions[fnName].size() > 0
+          ) {
+            this->onAllocationFunctionCall(inputMethodsSet[argName], fnName); 
+          }
+          return; 
+        }
+
+      }
+    }
+    
+    // for pointer aliasing
+    std::string lval = dataflow::variable(receivingValue); 
+    std::string rval = dataflow::variable(valueToStore);  
+
+    /*
+    LLVM identifiers either start with @ (global identifier) or % (local identifier)
+    Docs: https://llvm.org/docs/LangRef.html#identifiers 
+    For (temporary decision) simplicity sake, @'s will be replaced with %, though in the future the variable naming system may entirely change to remove %'s and @'s 
+
+    */
+    if (lval[0] == '@') {
+      lval[0] = '%'; 
+    }
+
+    if (rval[0] == '@') {
+      rval[0] = '%'; 
+    }
+
+    if (lval[0] == rval[0] && lval[0] == '%') {
+      aliasedVars[lval] = rval; 
+    }
+  
+
+  
+  }
+  else if (auto Call = dyn_cast<CallInst>(instruction)) {
+    for (unsigned i = 0; i < Call->getNumArgOperands(); ++i) {
+        Value *argument = Call->getArgOperand(i);
+        std::string argName = dataflow::variable(argument);
+
+        argName = aliasedVars[argName];
+        while (argName.size() > 1 && dataflow::isNumber(argName.substr(1))) {
+          argName = aliasedVars[argName]; 
+        }
+
+
+        // * i THINK this is what distinguishes "real" variables defined in the program from constants and other irrelevant llvm stuff 
```

Comment:
Not needed in this PR, but we should open issues on this repo to resolve both this comment and the one on line 207. For both, the issue should suggest writing a test to check whether these are correct.

Reply:
Upon testing and seeing what `argName` is at runtime I now realize that it's more apt to check if `argName` is an empty string (empty string meaning it's not in the `aliasedVars` map). `aliasedVars` filters out irrelevant LLVM IR data by virtue of only storing variables names and intermediate variables (e.g., %str, %1, %2). 
Now for the comment on line 207 I'm still unsure, so that'll probably get its own issue, but I don't think we'll need to go back on the comment for line 192 as I'm confident that we just need to check if `argName` is an empty string